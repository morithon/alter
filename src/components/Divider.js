import React from 'react';
import {View, StyleSheet} from 'react-native';

const styles = StyleSheet.create({
	divider: {
		flex: 1,
		backgroundColor: 'white'
	},
	success: {
		backgroundColor: 'green'
	},
	failure: {
		backgroundColor: 'red'
	}
});

type DividerProps = {
	isSuccess: boolean;
};

const getStyle = isSuccess => isSuccess ? styles.success : styles.failure;

const Divider = ({isSuccess}: DividerProps) => <View style={[styles.divider, getStyle(isSuccess)]}/>;

export default Divider;