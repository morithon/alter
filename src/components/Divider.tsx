import React from 'react';
import {StyleSheet, View} from 'react-native';
import { withFadeInAndOutAnimation, FadeInAndOutAnimationProps } from './withFadeInAndOutAnimation';
import { green, red } from '../styles/colors';

const styles = StyleSheet.create({
	divider: {
		flex: 1
	},
	success: {
		backgroundColor: green
	},
	failure: {
		backgroundColor: red
	}
});

interface DividerProps {
	isSuccess?: boolean;
};

const getStyle = (isSuccess: boolean | undefined) => {
	if (isSuccess === true) {
		return styles.success;
	} else if (isSuccess === false) {
		return styles.failure;
	}

	return;
};

const Divider = ({isSuccess}: DividerProps) => <View style={[styles.divider, getStyle(isSuccess)]}/>;

const shouldStartAnimation =
	(prevProps: FadeInAndOutAnimationProps<DividerProps>, props: FadeInAndOutAnimationProps<DividerProps>) =>
		typeof props.isSuccess === 'boolean' && prevProps.isSuccess !== props.isSuccess;

export default withFadeInAndOutAnimation<DividerProps>(Divider, shouldStartAnimation);