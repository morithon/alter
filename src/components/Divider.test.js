import React from 'react';
import Divider from './Divider';

import renderer from 'react-test-renderer';

it('render matches snapshot', () => {
  expect(renderer.create(<Divider isSuccess/>)).toMatchSnapshot();
  expect(renderer.create(<Divider isSuccess={false}/>)).toMatchSnapshot();
});