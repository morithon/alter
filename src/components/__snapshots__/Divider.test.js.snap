// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render matches snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "flex": 1,
      },
      Object {
        "backgroundColor": "green",
      },
    ]
  }
/>
`;

exports[`render matches snapshot 2`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "white",
        "flex": 1,
      },
      Object {
        "backgroundColor": "red",
      },
    ]
  }
/>
`;
